tensorboard/plugins/histogram/summary.py

```diff
@@ -0,0 +1,201 @@
+        right_edges = edges[1:]
+        return tf.transpose(tf.stack(
+          [left_edges, right_edges, bucket_counts]))
+
+      def when_singular():
+        center = min_
+        bucket_starts = tf.stack([center - 0.5])
+        bucket_ends = tf.stack([center + 0.5])
+        bucket_counts = tf.stack([tf.cast(tf.size(data), tf.float64)])
+        return tf.transpose(
+          tf.stack([bucket_starts, bucket_ends, bucket_counts]))
+
+      return tf.cond(is_singular, when_singular, when_nonsingular)
+
+    return tf.cond(is_empty, when_empty, when_nonempty)
+
+
+def op(name,
+       data,
+       bucket_count=None,
+       display_name=None,
+       description=None,
+       collections=None):
+  """Create a histogram summary op.
+
+  Arguments:
+    name: A unique name for the generated summary node.
+    data: A `Tensor` of any shape. Must be compatible with `float64`.
+    bucket_count: Optional positive `int` or scalar `int32` `Tensor`.
+    display_name: Optional name for this summary in TensorBoard.
+      Defaults to `name`.
+    description: Optional long-form description for this summary.
+      Markdown is supported.
+    collections: Optional list of graph collections keys. The new
+      summary op is added to these collections. Defaults to
+      `[Graph Keys.SUMMARIES]`.
+
+  Returns:
+    A TensorFlow summary op.
+  """
+  if display_name is None:
+    display_name = name
+  summary_metadata = metadata.create_summary_metadata(
+    display_name=display_name, description=description)
+  with tf.name_scope(name):
+    tensor = _buckets(data, bucket_count=bucket_count)
+    return tf.summary.tensor_summary(name='histogram_summary',
+                                     tensor=tensor,
+                                     collections=collections,
+                                     summary_metadata=summary_metadata)
+
+
+def pb(name, data, bucket_count=None, display_name=None, description=None):
+  """Create a histogram summary protobuf.
+
+  Arguments:
+    name: A unique name for the generated summary, including any desired
+      name scopes.
+    data: A `np.array` or array-like form of any shape. Must have type
+      compatible with `float`.
+    bucket_count: Optional positive `int`.
+    display_name: Optional name for this summary in TensorBoard.
+      Defaults to `name`.
+    description: Optional long-form description for this summary.
+      Markdown is supported.
+  """
+  if bucket_count is None:
+    bucket_count = BUCKETS
+  if bucket_count <= 0:
+    raise ValueError('assertion failed: "bucket_count" must be positive, '
+                     'but was %s' % bucket_count)
+  data = np.array(data).flatten().astype(float)
+  if data.size == 0:
+    buckets = np.array([]).reshape((0, 3))
+  else:
+    min_ = np.min(data)
+    max_ = np.max(data)
+    range_ = max_ - min_
+    if range_ == 0:
+      center = min_
+      buckets = np.array([[center - 0.5, center + 0.5, float(data.size)]])
```

Comment:
I think it's better to have it inlined. IMO, it should be possible to understand the specific behavior of a funciton without looking at the module docstring. Also, it's somewhat less likely to drift, because people are used to updating fn docstrings, not module docstrings.